tests/spark/autologging/conftest.py

```diff
@@ -49,3 +48,16 @@ def format_to_file_path(spark_session):
         for data_format, file_path in res.items():
             df.write.option("header", "true").format(data_format).save(file_path)
         yield res
+
+
+@pytest.fixture(autouse=True)
+def tear_down():
+    yield
+
+    # Clear cached table infos. When the datasource event from Spark arrives but there is no
+    # active run (e.g. the even comes with some delay), MLflow keep them in memory and logs them to
+    # the next **and any successive active run** (ref: PR #4086).
+    # However, this behavior is not desirable during tests, as we don't want any tests to be
+    # affected by the previous test. Hence, this fixture is executed on every test function
+    # to clear the accumulated table infos stored in the global context.
+    clear_table_infos()
```

Comment:
Can we apply `@pytest.fixture(autouse=True)` on `clear_table_infos`?

Reply:
Do you mean putting pytest annotation to the original def in `_spark_autologging.py`?

Reply:
I see, `clear_table_infos` is imported from other module. Never mind.